### PR TITLE
More reactive SQL client connection options

### LINKED ISSUE #11149: Expose reconnection properties for reactive datasources
**Description**
The reactive datasource drivers from Vert.x support reconnection:
https://vertx.io/docs/vertx-mysql-postgresql-client/java/#_configuring_reconnections
However, this configuration seems unreachable using standard Quarkus configuration methods. (or undocumented)

Providing the ability to define reconnect properties would be hugely useful in order to gracefully handle database failovers and general network blips.

### PR DESCRIPTION
**Pull request by tsegismont**
Added reconnectAttempts, reconnectInterval and idleTimeout props to reactive datasource config	

Resolves #11149

### CODE REVIEW

File: extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java [hunk 1]
```diff
@@ -94,4 +95,22 @@
...
+    /**
+     * The interval between reconnection attempts when a pooled connection cannot be established on first try.
+     */
+    @ConfigItem(defaultValue = "PT1S")
+    public Duration reconnectInterval;
+
+    /**
+     * The maximum time without data written to or read from a connection before it is removed from the pool.
+     */
+    @ConfigItem(defaultValue = "PT0S")
```
Comment (gsmet) [inline extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java:114]: Hmmm, it looks like this one should more be `Optional`?

**Review by gsmet — CHANGES_REQUESTED** [review]
Added a suggestion. Happy to discuss it as I'm not really sure.

**Comment by gsmet** [comment]
@machi1990 we have a doc generation issue with the duration note. We include it twice in the datasource.doc now that we have it included here too. I don't exactly know how we could fix that as these things are generated totally independently.

File: extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java (hunk 1 above)
Comment (tsegismont) [reply extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java]: Done

**Comment by gsmet** [comment]
I will try to figure out how to fix the doc issue.

**Comment by tsegismont** [comment]
Thanks

Le jeu. 20 août 2020 à 16:07, Guillaume Smet <notifications@github.com> a
écrit :

> I will try to figure out how to fix the doc issue.
>

**Comment by gsmet** [comment]
I added a quick (and dirty) fix for the doc issue. Let's see what CI says.

**Review by gsmet — APPROVED** [review]

**Comment by machi1990** [comment]
> @machi1990 we have a doc generation issue with the duration note. We include it twice in the datasource.doc now that we have it included here too. I don't exactly know how we could fix that as these things are generated totally independently.

@gsmet @tsegismont sorry only seeing this comment now. Good thing we had a fix in place :-)

**Pull request merged**